### PR TITLE
Take into account queue length in autoscaling

### PR DESCRIPTION
**Pull request by ericl**
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.

### CODE REVIEW

File: python/ray/monitor.py [hunk 1]
```diff
@@ -357,6 +360,7 @@ def run(self):
         try:
             self._run()
         except Exception:
+            logger.exception("Error in monitor loop")
```
Comment (ericl) [inline python/ray/monitor.py]: If you don't do this, the error message is lost forever trying to terminate nodes.

File: src/ray/raylet/scheduling_queue.cc [hunk 2]
```diff
@@ -131,8 +131,11 @@ const Task &SchedulingQueue::GetTaskOfState(const TaskID &task_id,
 }
 
 ResourceSet SchedulingQueue::GetResourceLoad() const {
-  // TODO(atumanov): consider other types of tasks as part of load.
-  return ready_queue_->GetCurrentResourceLoad();
+  auto load = ready_queue_->GetCurrentResourceLoad();
+  // Also take into account infeasible tasks so they show up for autoscaling
```
Comment (edoakes) [inline src/ray/raylet/scheduling_queue.cc]:
```suggestion
  // Also take into account infeasible tasks so they show up for autoscaling.
```

File: python/ray/tests/test_autoscaler.py [hunk 3]
```diff
@@ -142,45 +142,56 @@ def terminate_node(self, node_id):
 class LoadMetricsTest(unittest.TestCase):
     def testUpdate(self):
         lm = LoadMetrics()
-        lm.update("1.1.1.1", {"CPU": 2}, {"CPU": 1})
+        lm.update("1.1.1.1", {"CPU": 2}, {"CPU": 1}, {})
```
Comment (edoakes) [inline python/ray/tests/test_autoscaler.py]: Add some test cases for the new metric

File: python/ray/monitor.py (hunk 1 above)
Comment (edoakes) [reply python/ray/monitor.py]: Does this mean nodes probably weren't cleaned up? If so, would be good to print that in the error message.

Comment (edoakes) [inline python/ray/monitor.py]:
```suggestion
            logger.exception("Error in monitor loop.")
```

**Review by edoakes — APPROVED** [review]
LGTM

File: python/ray/tests/test_autoscaler.py (hunk 3 above)
Comment (ericl) [reply python/ray/tests/test_autoscaler.py]: Yeah there are a couple entries below in testLoadMessages

**Comment by ericl** [comment]
Merging so I can test this for real on compiled wheels -- had some issues trying to set it up before.

**Comment by ericl** [comment]
I just tested this on a real cluster and it works as expected.

**Pull request merged**